### PR TITLE
Maven: Add support for authors

### PR DESCRIPTION
**Pull request by bs-ondem**
Extract information about the authors of a package from the pom.xml
definition file and make it available in the metadata of packages
and project.

Since the pom.xml file does not contain any direct information
on authors, the authors field in Package/Project is filled
with the help of organization or developers tag.

### CODE REVIEW

**Review by KorSin — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt [hunk 1]
```diff
@@ -122,6 +122,22 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
...
+        fun parseOrganizations(mavenProject: MavenProject) =
+            sortedSetOf<String>().apply {
+                mavenProject.organization?.let {
+                    if (!it.name.isNullOrEmpty()) {
+                        add(it.name)
+                    }
+                }
+                mavenProject.developers.mapNotNull {
+                    if (!it.organization.isNullOrEmpty()) {
+                        add(it.organization)
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/utils/MavenSupport.kt:134]: See https://github.com/oss-review-toolkit/ort/pull/3627#issuecomment-778342059.

File: analyzer/src/funTest/assets/projects/external/jgnash-core-expected-output.yml [hunk 2]
```diff
@@ -99,6 +99,10 @@ packages:
     path: ""
 - id: "Maven:com.fasterxml:classmate:1.3.0"
   purl: "pkg:maven/com.fasterxml/classmate@1.3.0"
+  authors:
+  - "Brian Langel"
+  - "Tatu Saloranta"
+  - "fasterxml.com"
```
Comment (sschuberth) [inline analyzer/src/funTest/assets/projects/external/jgnash-core-expected-output.yml:105]: This (and similar entries below) stem from `organization -> name` and don't seem useful in the *author* context to me. I think we should experiment with disregarding that field, and looking what else it makes disappear from these expected results. If all disappearing results are in the domain name form, I believe should indeed not look at  `organization -> name`.

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt [hunk 3]
```diff
@@ -122,6 +122,22 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
...
         }
 
+        fun parseOrganizations(mavenProject: MavenProject) =
+            sortedSetOf<String>().apply {
+                mavenProject.organization?.let {
+                    if (!it.name.isNullOrEmpty()) {
+                        add(it.name)
+                    }
+                }
+                mavenProject.developers.mapNotNull {
```
Comment (oheger-bosch) [inline analyzer/src/main/kotlin/managers/utils/MavenSupport.kt:132]: The usage of `mapNotNull` is a bit strange here, as nothing is done with the result of the mapping. I assume, we want something like:

```
val authors = mavenProject.developers.mapNotNull {
  it.organization?.takeUnless { it.isEmpty() } ?: it.name
}
addAll(authors)

```
Or `authors` could even be inlined.

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt [hunk 4]
```diff
@@ -122,6 +122,22 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
             }
         }
 
+        fun parseOrganizations(mavenProject: MavenProject) =
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/MavenSupport.kt:125]: As this not only parses organization tags, we probably should name this function *for* what information is parsed, and use `parseAuthors` like we already do for `Npm` and `Pub`.

File: analyzer/src/funTest/assets/projects/external/jgnash-core-expected-output.yml (hunk 2 above)
Comment (sschuberth) [reply analyzer/src/funTest/assets/projects/external/jgnash-core-expected-output.yml:105]: I just gave this a try. While omitting `organization -> name` gets rid of entries like `"fasterxml.com"`, `"com.mchange"`, and some "duplicates" like `"XStream"` (where we already have `"XStream Committers"`), `"JBoss by Red Hat"` (where we already have `"JBoss.org"`), or `"The Apache Software Foundation"` (where we already have `"Apache Software Foundation"`), it also removes valid entries like `"MetaStuff Ltd."`, `"JUnit"`, or `"Pivotal Software, Inc."`.

So bottom line, I think we need to keep looking at `organization -> name`. However, while looking at these results I'm now wondering whether we should further distinguish between organizations and persons in the `authors` field (similar to [like SPDX does](https://spdx.github.io/spdx-spec/3-package-information/#36-package-originator)), or whether it's fine to mix these. If we mix these here, we later have no chance anymore to separate them, e.g. when *generating* SPDX.

Comment (bs-ondem) [reply analyzer/src/funTest/assets/projects/external/jgnash-core-expected-output.yml:105]: I'm also not happy about the "domain name form" entries or duplicates. But if you omit `organization -> name`, useful entries are also omitted. That's why I'm a little overwhelmed by how I should best deal with it. But even if we distinguish between organizations and persons, we will have duplicates or domain-like entries.

Comment (sschuberth) [reply analyzer/src/funTest/assets/projects/external/jgnash-core-expected-output.yml:105]: Yeah, let's just go with the current approach and eventually fine-tune the logic later.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**